src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/NullApplicationPartFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// An <see cref="ApplicationPartFactory"/> that produces no parts.
+    /// <para>
+    /// This factory may be used to to preempt Mvc's default part discovery allowing for custom configuration at a later stage.
+    /// </para>
+    /// </summary>
+    public class NullApplicationPartFactory : ApplicationPartFactory
```

Comment:
nit: I'm struggling with `Null` here. We need to find a better name, we should brainstorm

Reply:
Sure. We have precedence for `NullThing` in BCL \ AspNetCore though.

Reply:
Fair enough, I don't really have a better name.